orbit/estimators/stan_estimator.py

```diff
@@ -139,44 +131,48 @@ def fit(
-                iter=self._num_iter_per_chain,
-                warmup=self._num_warmup_per_chain,
-                chains=self.chains,
-                n_jobs=self.cores,
-                # fall back to default if not provided by model payload
-                init=init_values,
-                seed=self.seed,
-                algorithm=self.algorithm,
-                control=self.stan_mcmc_control,
-                **self._stan_mcmc_args,
-            )
+        compiled_mod = get_compiled_stan_model(model_name)
+        # check https://mc-stan.org/cmdstanpy/api.html#cmdstanpy.CmdStanModel.sample
+        # for additional args
+        stan_mcmc_fit = compiled_mod.sample(
+            data=data_input,
+            iter_sampling=self._num_sample_per_chain,
+            iter_warmup=self._num_warmup_per_chain,
+            chains=self.chains,
+            parallel_chains=self.cores,
+            inits=init_values,
+            seed=self.seed,
+            **self._stan_mcmc_args,
+        )
 
-        posteriors = stan_mcmc_fit.extract(pars=model_param_names, permuted=False)
+        stan_extract = stan_mcmc_fit.stan_variables()
+        posteriors = {
+            param: stan_extract[param] for param in model_param_names + ["loglk"]
+        }
 
+        # TODO: cmdstan is the API now; does it preserve the chain information
+        # by default?
         # todo: move dimension cleaning function to the model directly
         # flatten the first two dims by preserving the chain order
-        for key, val in posteriors.items():
-            if len(val.shape) == 2:
-                # here `order` is important to make samples flattened by chain
-                posteriors[key] = val.flatten(order="F")
-            else:
-                posteriors[key] = val.reshape((-1, *val.shape[2:]), order="F")
+        # for k, v in posteriors.items():
+        #     print(k)
+        #     print(v.shape)
+        #     if len(val.shape) == 2:
+        #         # here `order` is important to make samples flattened by chain
+        #         posteriors[k] = v.flatten(order="F")
+        #     else:
+        #         posteriors[k] = v.reshape((-1, *v.shape[2:]), order="F")
 
         # extract `log_prob` in addition to defined model params
         # to make naming consistent across api; we move lp along with warm up lp to `training_metrics`
         # model_param_names_with_lp = model_param_names[:] + ['lp__']
-        loglk = stan_mcmc_fit.extract(pars=["loglk"], permuted=True)["loglk"]
+        loglk = posteriors["loglk"]
         training_metrics = {"loglk": loglk}
-        training_metrics.update(
-            {"log_posterior": stan_mcmc_fit.get_logposterior(inc_warmup=True)}
-        )
+        # log_posterior is not supported in cmdstanpy
+        # training_metrics.update(
+        #     {"log_posterior": stan_mcmc_fit.get_logposterior(inc_warmup=True)}
+        # )
         training_metrics.update({"sampling_temperature": sampling_temperature})
```

Comment:
nit: we can create `training_metrics` at once instead of in two steps.
```
training_metrics = {
    "loglk": loglk,
    "sampling_temperature": sampling temperature,
}
```